### DISCUSSION TITLE
calling custom field in a custom script.

### DISCUSSION THREAD

**Discussion by john2893** [netbox-community/netbox#13588] [Help Wanted!] [answered]
I added a custom field in Netbox via the web.
```
Name = testing
Label = Testing
Type - multiple selection
Content types = Zone
Assigned Models = netbox_dns | dns 
default value = None 
Choices = A, B
```
I am trying to pull this custom field data in using python, I can get the zone, view, records info but just not this one.
I tried this: but this does not work.
```
zones = Zone.objects.filter(view=view, status__in=Zone.ACTIVE_STATUS_LIST,custom_field_data="testing")
for zone in zones:
print (zone) 
```
I'm sure its simple but just can't figure this out.
how do i do this? I'm kind of stuck

Is there a module like : 
`from netbox_dns.models import View, Zone, Record` that I can use to call the custom fields ?

Currently I have this script that works to display: Zone, records and view name:

`#!/usr/bin/env python3

from pathlib import Path
from netbox_dns.models import View, Zone, Record
from extras.scripts import *
from extras.scripts import Script, StringVar, BooleanVar
from jinja2 import Environment, DictLoader

     for view in views:
            zones = Zone.objects.filter(view=view, status__in=Zone.ACTIVE_STATUS_LIST)
            if len(zones):
                self.log_info(f"Exporting zones for view '{view.name}'")
                self.export_zones(zones, view.name, export_path)

        zones = Zone.objects.filter(view__isnull=True, status__in=Zone.ACTIVE_STATUS_LIST)
        if len(zones):
            self.log_info("Exporting zones without a view")
            self.export_zones(zones, data["default_view_name"], export_path)
  
        for zone in zones:
            self.log_info(f"Exporting zone {zone}")
            print (zone)

            records = Record.objects.filter(zone=zone, status__in=Record.ACTIVE_STATUS_LIST)
            print ("RECORDS")
            print (records)
            zone_data = self.template.render({"zone": zone, "records": records})
            print ("ZONE DATA ")
            print (zone_data)`

**Comment by candlerb** [comment, marked as answer]
`custom_field_data` is a single JSON object containing *all* custom fields.  So there's no need to filter on it:

```
zones = Zone.objects.filter(view=view, status__in=Zone.ACTIVE_STATUS_LIST)
for zone in zones:
    print (zone, zone.custom_field_data)
```

(assuming that this plugin model has been defined to permit custom fields)

In this case, the content of custom_field_data would look like `{"testing":"A"}`

**Reply by john2893** [reply]
Thank you very much!!

**Comment by zamanrahimi** [comment]


**Reply by candlerb** [reply]
@zamanrahimi, there are a several problems with that posting.

* You hijacked someone else's thread to ask an unrelated question. That's considered very impolite and makes the archived discussions hard to search and follow. Please start a new thread.
* You addressed it to an individual, rather than to the community. That is also very impolite, as it sounds like you are demanding a particular individual to respond to you personally.
* I suspect you are Netbox "user", not "customer". This might just be a question of language (and of course, it's fine if English is not your first language - and welcome!) However, if you are indeed a customer of [Netbox Labs](https://netboxlabs.com/pricing/) then there will be a separate support channel that you should use.
* You have not supplied any example of the code or data which causes the problem, or explained what exactly your script is trying to do. Therefore, we're not going to be able to help. See https://catb.org/~esr/faqs/smart-questions.html#intro for good advice on how to ask questions in a way that helps people to identify the problem and in turn help you.

**Comment by zamanrahimi** [comment]


**Reply by candlerb** [reply]
You can - look for the three dots menu, there's an Edit option. But I do suggest that you start a new discussion:
https://github.com/netbox-community/netbox/discussions
Click "New Discussion"
